aws/rds_test.go

```diff
@@ -0,0 +1,92 @@
+package aws
+
+import (
+	"testing"
+	"time"
+
+	"strings"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/rds"
+
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func waitUntilRdsCreated(svc *rds.RDS, name *string) error {
+	input := &rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: name,
+	}
+
+	for i := 0; i < 240; i++ {
+		instance, err := svc.DescribeDBInstances(input)
+		status := instance.DBInstances[0].DBInstanceStatus
+
+		// If SkipFinalSnapshot = false on delete, should also wait for "backing-up" also to finish
+		if *status != "creating" {
+			return nil
+		}
+
+		if err != nil {
+			return err
+		}
+
+		time.Sleep(1 * time.Second)
+		logging.Logger.Debug("Waiting for RDS to be created")
+	}
+
+	return RdsDeleteError{}
+}
+
+func createTestRDSInstance(t *testing.T, session *session.Session, name string) {
+	svc := rds.New(session)
+	params := &rds.CreateDBInstanceInput{
+		AllocatedStorage:     awsgo.Int64(5),
+		DBInstanceClass:      awsgo.String("db.m5.large"),
+		DBInstanceIdentifier: awsgo.String(name),
+		Engine:               awsgo.String("postgres"),
+		MasterUsername:       awsgo.String("gruntwork"),
+		MasterUserPassword:   awsgo.String("password"),
+	}
+
+	_, err := svc.CreateDBInstance(params)
+
+	if err != nil {
+		assert.Fail(t, err.Error())
+	}
```

Comment:
NIT: I know you got this from other code, but more recently we learned that we can replace this whole if condition with `require.NoError(t, err)`. The import for `require` is `"github.com/stretchr/testify/require"`.